doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/TagDocProcessor.kt

```diff
@@ -0,0 +1,303 @@
+     * @param documentable The Documentable beloning to the current tag.
+     * @param filteredDocumentables Filtered documentables by [filterDocumentables], which, by default, filters the
+     *  documentables to have a supported tag in them. The [DocumentableWrapper]s in this map are the same objects as in [allDocumentables], just a subset.
+     * @param allDocumentables All documentables in the source set.
+     *
+     * @return A [String] that will replace the tag with content entirely.
+     */
+    abstract fun processBlockTagWithContent(
+        tagWithContent: String,
+        path: String,
+        documentable: DocumentableWrapper,
+        filteredDocumentables: Map<String, List<DocumentableWrapper>>,
+        allDocumentables: Map<String, List<DocumentableWrapper>>,
+    ): String
+
+    /**
+     * Process an inline tag with content ([tagWithContent]) into whatever you like.
+     *
+     * @param tagWithContent Tag with content to process.
+     *  For example: `{@tag Some content}`
+     *  Contains {} and tag.
+     * @param path The path of the doc where the tag is found.
+     * @param documentable The Documentable beloning to the current tag.
+     * @param filteredDocumentables Filtered documentables by [filterDocumentables], which, by default, filters the
+     *  documentables to have a supported tag in them. The [DocumentableWrapper]s in this map are the same objects as in [allDocumentables], just a subset.
+     * @param allDocumentables All documentables in the source set.
+     *
+     * @return A [String] that will replace the tag with content entirely.
+     */
+    abstract fun processInlineTagWithContent(
+        tagWithContent: String,
+        path: String,
+        documentable: DocumentableWrapper,
+        filteredDocumentables: Map<String, List<DocumentableWrapper>>,
+        allDocumentables: Map<String, List<DocumentableWrapper>>,
+    ): String
+
+    /**
+     * This is the main function that will be called by the [DocProcessor].
+     * It will process all documentables found in the source set and replace all
+     * supported tags ([tagIsSupported]) and their content with the result of [processBlockTagWithContent]
+     * and [processInlineTagWithContent].
+     * The processing will continue until there are no more supported tags found in all
+     * filtered documentables. This means that recursion (a.k.a tags that create other supported tags)
+     * is possible. However, there is a limit to prevent infinite recursion ([ProcessDocsAction.Parameters.processLimit]).
+     */
+    override fun process(
+        parameters: ProcessDocsAction.Parameters,
+        documentablesByPath: Map<String, List<DocumentableWrapper>>,
+    ): Map<String, List<DocumentableWrapper>> {
+        // Convert to mutable
+        val allDocumentables = documentablesByPath.toMutable()
+
+        // Filter documentables
+        val filteredDocumentables = allDocumentables
```

Comment:
This calls the function `filterDocumentables` that can be overridden by the user. So that's why it might look like it appears more often. 
I did filter for tags twice since the filter function could be overridden. Fixed that.